packages/components/src/internal/components/search/EntityFieldFilterModal.tsx

```diff
@@ -125,15 +126,23 @@ export const EntityFieldFilterModal: FC<Props> = memo(props => {
         [api, entityDataType, skipDefaultViewCheck, showAllFields]
     );
 
+    const allowFaceting = useMemo(() => {
+        return activeField?.allowFaceting() && activeField?.jsonType === 'string';
```

Comment:
Are the `jsonType === 'string'` checks here and below at line 140 still just temporary until Matt's current story is complete? If so, should a TODO comment be added here about removing the check at that time?

Reply:
It's not related to the sql technical issue, but rather, we are intentional scoping it down since other field types don't seem to benefit much from it. will add comment to clarify